compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -830,23 +830,23 @@ object Types {
      *  def o: Outer
      *  <o.x.type>.widen = o.C
      */
-    @tailrec final def widen(implicit ctx: Context): Type = widenSingleton match {
```

Comment:
It was noise. I was against adding it, but it got merged before I could comment.

As I wrote then: Add a @tailrec only if it is important that the method is in fact tail-recursive. Either it is super-performance critical or can potentially recurse deeply. Do not add @tailrec just because the method happens to be tail recursive.